src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -95,11 +95,7 @@ internal partial class AuthenticationHelper
                 return null;
             }
 
-            string opaque;
-            if (!digestResponse.Parameters.TryGetValue(Opaque, out opaque))
-            {
-                return null;
-            }
+            string opaque = digestResponse.Parameters.ContainsKey(Opaque) ? digestResponse.Parameters[Opaque] : null;
```

Comment:
Technically speaking, this will do 2 lookups. I would suggest to keep the `!TryGetValue` condition and set explicitly `opaque = null`.

Reply:
The code was modeled by existing line below. Should I fix the realm lookup as well?

Reply:
Hmmm, I don't have strong opinion. I would update both, but opinions might differ ...

Reply:
Yeah, please change the realm and opaque pattern to TryGetValue..